### PR TITLE
Feature/injections 

### PR DESCRIPTION
**Pull request by j-d-carmichael**
This injector merges over files from tpl files in an opt'in fashion.

Example:

1. The original tpl repo: https://github.com/acr-lfr/generate-it-typescript-server 
2. The injector example repo: https://github.com/j-d-carmichael/test-injection-repo 
3. An example dummy API ( https://github.com/j-d-carmichael/api-dummy/tree/typeorm ):
   - Generated with the generate-it-typescript-server
   - .nodegenrc file contains the inection block
   - The injection references the aforementioned test-injeciton-repo

What happens when the generate-server on the dummy api:

1. normal ts server tpl is downloaded
2. sees the injection block in the rc file (https://github.com/j-d-carmichael/api-dummy/blob/typeorm/.nodegenrc#L7)
3. creates a new "_merged" folder (a copy of the previously downloaded tpl, minus the .git [this is not needed and created issues])
4. now it loops over all injection sources in the injection block
   - downloads the source
   - merges the source into the _merged folder
5. Finally, the injection.init returns the newly found templates system filepath
6. Genetate-it core continues oblivious to the hack.

The output of the example:
1. Everything is the same as the base template except any overwrites from the injection.
   - src/http/index.ts https://github.com/j-d-carmichael/api-dummy/blob/typeorm/src/http/index.ts#L32  the additional injections have been snipered out via the code injection

All previous tests work just the same. It is as if it was never there.

### CODE REVIEW

**Comment by j-d-carmichael** [comment]
@p-mcgowan   @SergiAcrontum   @acrontum-sanchez 

i'm happy with the naming on this now - could i get a review on the concept/code?

File: src/interfaces/NodegenRc.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
-  [helperName: string]: any
+  [helperName: string]: any;
 }
 
-export interface InjectionOverwrites {
-  targetDir: string,
-  url: string
+export interface Injection {
+  destination?: string;
```
Comment (j-d-carmichael) [inline src/interfaces/NodegenRc.ts:9]: this is redundant at the moment - should be removed, hints at recipes but this is not what this feature does

**Comment by p-mcgowan** [comment]
Yeah it looks good, I want to run it through a few projects and give it a spin but I don't see any issues (maybe change merge base to dev).

We use "Inject" / "injections" quite often in the project since most of what the code does is some kind of inject, I'm wondering if it makes sense using "overwrite / override" in the naming somewhere, since it's merging the folder structures but overwriting the files?

**Comment by j-d-carmichael** [comment]
> Yeah it looks good, I want to run it through a few projects and give it a spin but I don't see any issues (maybe change merge base to dev).
> 
> We use "Inject" / "injections" quite often in the project since most of what the code does is some kind of inject, I'm wondering if it makes sense using "overwrite / override" in the naming somewhere, since it's merging the folder structures but overwriting the files?

Yea i'm not 100% on injection either. Could be "mergeWithTemplates: []"  or something?

I actually moved away from overwrite as technically, you can use this to just inject new files and not overwrite or a mixture of both.

**Comment by j-d-carmichael** [comment]
It would also be nice to be able to delete tpl files before running the template rendering.

Example:

1. Use the base ts tpl
2. Merge/inject/overwrite in a new tpl which creates a new AccessToken middleware handler
3. Delete now redundant one:  https://github.com/acr-lfr/generate-it-typescript-server/blob/master/src/services/AccessTokenService.ts

Not sure if this is a feature for another PR after this one is completed though

**Review by p-mcgowan — APPROVED** [review]
Yeah definitely gets a bit complicated when designing something that supports deletions / extension / overwriting.

Code looks fine and doesn't affect current projects. Not sure what's up with the linter but doesn't seem to catch much, however if lint passes then it's all good.

Can you merge to dev instead of main?

**Pull request merged**